### DISCUSSION TITLE
README on puma with workers

### DISCUSSION THREAD

**Discussion by majkelcc** [bensheldon/good_job#812] [Q&A] [answered]
`README.rb` contains this note on using GoodJob with Puma:

> When running Puma with workers (WEB_CONCURRENCY > 0) or another process-forking web server, GoodJob's threadpool schedulers should be stopped before forking, restarted after fork, and cleanly shut down on exit. Stopping GoodJob's scheduler pre-fork is recommended to ensure that GoodJob does not continue executing jobs in the parent/controller process.

Does this apply only to the situation when GoodJob is configured to run jobs within the web server process (and we use Puma workers)?

**Comment by bensheldon** [comment, marked as answer]
@majkelcc Yes, you will only need to call those GoodJob methods in the `config/puma.rb` hooks _if you are running GoodJob in `:async` mode_. 

If you are using the standalone `$ good_job` process to execute jobs, you don't need to add those hooks.

Sorry about the slow reply!

**Reply by laptopmutia** [reply]
by standalone `$ good_job` did you mean with :external mode?

**Comment by majkelcc** [comment]
@bensheldon thanks for clarifying. I imagined this would be the usual case on production, so it could be worth to mention it in the README, but then again it probably doesn't hurt and it could be more common in the development env anyway.
